packages/block-library/src/gallery/style.scss

```diff
@@ -80,26 +78,44 @@
 		}
 	}
 
-	// Responsive fallback value, 2 columns
+	// On mobile and responsive viewports, we allow only 1 or 2 columns at the most.
 	& .blocks-gallery-image,
 	& .blocks-gallery-item {
-		width: calc(100% / 2 - 16px);
+		width: calc((100% / 2) - (#{ $grid-size-large } * 2));
 	}
 
 	&.columns-1 .blocks-gallery-image,
 	&.columns-1 .blocks-gallery-item {
-		width: calc(100% / 1 - 16px);
+		width: 100%;
 	}
 
+	// Beyond mobile viewports, we allow up to 8 columns.
 	@include break-small {
 		@for $i from 3 through 8 {
 			&.columns-#{ $i } .blocks-gallery-image,
 			&.columns-#{ $i } .blocks-gallery-item {
-				width: calc(100% / #{ $i } - 16px);
+				width: calc((100% / #{ $i }) - (#{ $grid-size-large } * #{ $i }));
 			}
 		}
 	}
 
+	// Unset the right margin on every rightmost gallery item to ensure center balance.
+	@for $j from 1 through 8 {
```

Comment:
I think even though this is scss, I still prefer giving variables a proper name like $columnCount. As far as I know, sass has block scope, so it shouldn't cause an issue.

Reply:
Done!